src/lib/util/VersionUpgradeManager.ts

```diff
@@ -75,6 +95,18 @@ export class VersionUpgradeManager {
 		return !configVersion || (semver.lt(configVersion, this.currentVersion));
 	}
 
+	/**
+	 * Depending on what version we're upgrading from, we may want to force the creation
+	 * of a back-up config, instead of only creating one in response to failure.
+	 * This method indicates whether that's the case.
+	 * @param configVersion
+	 * @returns true if a back-up should always be created, else false
+	 */
+	public versionNecessitatesBackup(configVersion: string): boolean {
```

Comment:
New method used in `Config.ts` to determine whether a backup is always necessary or merely situationally necessary.